token/program-2022/src/instruction.rs

```diff
@@ -29,6 +29,8 @@ pub enum TokenInstruction {
     /// Otherwise another party can acquire ownership of the uninitialized
     /// account.
     ///
+    /// Any extensions must be initialized before this instruction.
```

Comment:
nit: I think this comment and the similar ones for the `InitializeAccount` family could be dropped.  This is all true with the current set of extensions but may not be the case going forward.  More generally some extensions will need to be initialized prior to `InitializeMint`/`InitializeAccount` and some can be done later.